### PR TITLE
Make it possible to pass the auth on a per request basis

### PR DESCRIPTION
**Pull request by dstufft**
This allows end users to solve #23.

### CODE REVIEW

File: gidgethub/sansio.py [hunk 1]
```diff
@@ -145,9 +146,13 @@ def create_headers(requester: str, *,
     # accept: https://developer.github.com/v3/#current-version
     #         https://developer.github.com/v3/media/
     # authorization: https://developer.github.com/v3/#authentication
+    if oauth_token is not None and authorization is not None:
+        raise ValueError("Cannot pass both oauth_token and authorization")
     headers = {"user-agent": requester, "accept": accept}
     if oauth_token is not None:
         headers["authorization"] = f"token {oauth_token}"
+    if authorization is not None:
```
Comment (brettcannon) [inline gidgethub/sansio.py]: `elif`

File: docs/abc.rst [hunk 2]
```diff
@@ -98,7 +98,7 @@ experimental APIs without issue.
             Renamed from ``_sleep()``.
 
 
-    .. coroutine:: getitem(url: str, url_vars: Dict[str, str] = {}, *, accept=sansio.accept_format()) -> Any
+    .. coroutine:: getitem(url: str, url_vars: Dict[str, str] = {}, *, accept=sansio.accept_format(), authorization=None) -> Any
```
Comment (brettcannon) [inline docs/abc.rst]: I just realized that I left the type info out for `accept` which led you to leave it out for `authorization` everywhere. Now I have to decide if I need to go back and add the missing type info or remove it from the docs entirely.

File: gidgethub/abc.py [hunk 3]
```diff
@@ -28,11 +28,12 @@ def __init__(self, requester: str, *, oauth_token: str = None) -> None:
 
     async def _make_request(self, method: str, url: str,
                             url_vars: Dict[str, str], data: Any,
-                            accept) -> Tuple[Any, str]:
+                            accept, authorization=None) -> Tuple[Any, str]:
```
Comment (brettcannon) [inline gidgethub/abc.py]: Missing type info (as is `accept`, darn it).

File: gidgethub/test/test_abc.py [hunk 4]
```diff
@@ -61,6 +62,18 @@ def __init__(self, status_code=200, headers=DEFAULT_HEADERS, body=b''):
 
 
 @pytest.mark.asyncio
+async def test_custom_auth_headers():
+    """ Test that custom auth headers are created. """
```
Comment (brettcannon) [inline gidgethub/test/test_abc.py]: Leading and trailing whitespace within the docstring.

**Comment by brettcannon** [comment]
I just opened #27 to remove the type hints since mypy can't help keep them in check, so don't worry about that in your PR.

**Comment by Mariatta** [comment]
@dstufft This PR needs rebase 😊 
I'm interested to turn my [black_out](https://github.com/mariatta/black_out) bot into a GitHub app, and that effort is relying on this PR.
So I'm curious of the status of it.
Thanks!

**Comment by dstufft** [comment]
I forgot it existed, I probably won't get around to it anytime soon, feel free to take it over.

**Comment by Mariatta** [comment]
Thanks @dstufft! I'm closing this since we went with PR https://github.com/brettcannon/gidgethub/pull/62.

**Pull request closed without merging**